zkevm-circuits/src/evm_circuit/step.rs

```diff
@@ -0,0 +1,552 @@
+        *self as u64
+    }
+
+    pub(crate) fn iterator() -> impl Iterator<Item = Self> {
+        [
+            Self::BeginTx,
+            Self::STOP,
+            Self::ADD,
+            Self::MUL,
+            Self::DIV,
+            Self::SDIV,
+            Self::MOD,
+            Self::SMOD,
+            Self::ADDMOD,
+            Self::MULMOD,
+            Self::EXP,
+            Self::SIGNEXTEND,
+            Self::LT,
+            Self::SLT,
+            Self::ISZERO,
+            Self::AND,
+            Self::NOT,
+            Self::BYTE,
+            Self::SHL,
+            Self::SHR,
+            Self::SAR,
+            Self::SHA3,
+            Self::ADDRESS,
+            Self::BALANCE,
+            Self::ORIGIN,
+            Self::CALLER,
+            Self::CALLVALUE,
+            Self::CALLDATALOAD,
+            Self::CALLDATASIZE,
+            Self::CALLDATACOPY,
+            Self::CODESIZE,
+            Self::CODECOPY,
+            Self::GASPRICE,
+            Self::EXTCODESIZE,
+            Self::EXTCODECOPY,
+            Self::RETURNDATASIZE,
+            Self::RETURNDATACOPY,
+            Self::EXTCODEHASH,
+            Self::BLOCKHASH,
+            Self::COINBASE,
+            Self::TIMESTAMP,
+            Self::NUMBER,
+            Self::DIFFICULTY,
+            Self::GASLIMIT,
+            Self::CHAINID,
+            Self::SELFBALANCE,
+            Self::BASEFEE,
+            Self::POP,
+            Self::MLOAD,
+            Self::SLOAD,
+            Self::SSTORE,
+            Self::JUMP,
+            Self::JUMPI,
+            Self::PC,
+            Self::MSIZE,
+            Self::GAS,
+            Self::JUMPDEST,
+            Self::PUSH,
+            Self::DUP,
+            Self::SWAP,
+            Self::LOG,
+            Self::CREATE,
+            Self::CALL,
+            Self::CALLCODE,
+            Self::RETURN,
+            Self::DELEGATECALL,
+            Self::CREATE2,
+            Self::STATICCALL,
+            Self::REVERT,
+            Self::SELFDESTRUCT,
+            Self::ErrorInvalidOpcode,
+            Self::ErrorStackOverflow,
+            Self::ErrorStackUnderflow,
+            Self::ErrorWriteProtection,
+            Self::ErrorDepth,
+            Self::ErrorInsufficientBalance,
+            Self::ErrorContractAddressCollision,
+            Self::ErrorMaxCodeSizeExceeded,
+            Self::ErrorInvalidCreationCode,
+            Self::ErrorReverted,
+            Self::ErrorInvalidJump,
+            Self::ErrorReturnDataOutOfBound,
+            Self::ErrorOutOfGasConstant,
+            Self::ErrorOutOfGasPureMemory,
+            Self::ErrorOutOfGasSHA3,
+            Self::ErrorOutOfGasCALLDATACOPY,
+            Self::ErrorOutOfGasCODECOPY,
+            Self::ErrorOutOfGasEXTCODECOPY,
+            Self::ErrorOutOfGasRETURNDATACOPY,
+            Self::ErrorOutOfGasLOG,
+            Self::ErrorOutOfGasCALL,
+            Self::ErrorOutOfGasCALLCODE,
+            Self::ErrorOutOfGasDELEGATECALL,
```

Comment:
missing error  `ErrCodeStoreOutOfGas` or `ErrorOutOfGasCREATE2' represents it ?

Reply:
It's missing, thanks for catching! On my way adding this.

Reply:
Fixed in [`3e1241f`](https://github.com/appliedzkp/zkevm-circuits/pull/196/commits/3e1241f2316bb35b370a7602da2f059a545cb09e).